index.js

```diff
@@ -424,9 +429,9 @@
            '#' + stripNamespace(name);
   }
 
-  //  NullaryTypeWithUrl :: (String, Any -> Boolean) -> Type
-  function NullaryTypeWithUrl(name, test) {
-    return NullaryType(name, functionUrl(name), test);
+  //  NullaryTypeWithUrl :: (Type, String, Any -> Boolean) -> Type
+  function NullaryTypeWithUrl(parent, name, test) {
```

Comment:
I'd prefer `(name, parent, test)` for two reasons:

  - it makes the two constituents of the type's `test` method, `parent` and `test`, adjacent; and
  - having `name` first makes the call sites read better, in my view.